features/include/pcl/features/flare.h

```diff
@@ -0,0 +1,300 @@
+        */
+        inline void
+          setMarginThresh (float margin_thresh)
+        {
+          margin_thresh_ = margin_thresh;
+        }
+
+        /** \brief Get the percentage of the search tangent radius after which a point is considered part of the support.
+        *
+        * \return The percentage of the search tangent radius after which a point is considered part of the support.
+        */
+        inline float
+          getMarginThresh () const
+        {
+          return (margin_thresh_);
+        }
+
+
+        /** \brief Set min number of neighbours required for the computation of Z axis.
+        *
+        * \param[in] min_neighbors_for_normal_axis min number of neighbours required for the computation of Z axis.
+        */
+        inline void
+          setMinNeighboursForNormalAxis (int min_neighbors_for_normal_axis)
+        {
+          min_neighbors_for_normal_axis_ = min_neighbors_for_normal_axis;
+        }
+
+        /** \brief Get min number of neighbours required for the computation of Z axis.
+        *
+        * \return min number of neighbours required for the computation of Z axis.
+        */
+        inline int
+          getMinNeighboursForNormalAxis () const
+        {
+          return (min_neighbors_for_normal_axis_);
+        }
+
+
+        /** \brief Set min number of neighbours required for the computation of X axis.
+        *
+        * \param[in] min_neighbors_for_tangent_axis min number of neighbours required for the computation of X axis.
+        */
+        inline void
+          setMinNeighboursForTangentAxis (int min_neighbors_for_tangent_axis)
+        {
+          min_neighbors_for_tangent_axis_ = min_neighbors_for_tangent_axis;
+        }
+
+        /** \brief Get min number of neighbours required for the computation of X axis.
+        *
+        * \return min number of neighbours required for the computation of X axis.
+        */
+        inline int
+          getMinNeighboursForTangentAxis () const
+        {
+          return (min_neighbors_for_tangent_axis_);
+        }
+
+
+        /** \brief Provide a pointer to the dataset used for the estimation of X axis.
+        * As the estimation of x axis is negligibly affected by surface downsampling,
+        * this method lets to consider a downsampled version of surface_ in the estimation of x axis.  
+        * This is optional, if this is not set, it will only use the data in the
+        * surface_ cloud to estimate the x axis. 
+        * \param[in] cloud a pointer to a PointCloud 
+        */
+        inline void 
+          setSearchSampledSurface(const PointCloudInConstPtr &cloud)
+        {
+          sampled_surface_ = cloud;
+          fake_sampled_surface_ = false;
+        }
+
+        /** \brief Get a pointer to the sampled_surface_ cloud dataset. */
+        inline PointCloudInConstPtr 
+          getSearchSampledSurface() const
+        {
```

Comment:
You we're mimicking something you saw in the base class, which is good, but it doesn't mean the base class is doing it right. The reason I ask for the const reference is because with it you can do both

```cpp
const FLARELocalReferenceFrameEstimation<PointInT, PointOutT>::PointCloudInConstPtr& sampled_surface = flare.getSampledSurface (); //No copy of anything is made, no triggering of ref count
const FLARELocalReferenceFrameEstimation<PointInT, PointOutT>::PointCloudInConstPtr sampled_surface = flare.getSampledSurface (); //A copy is made, ref count is triggered.
```

With the old signature you can only do the second one, because you can't bind a reference to an rvalue.